codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -180,6 +181,27 @@ DefaultHttp2FrameStream newStream() {
         return new DefaultHttp2FrameStream();
     }
 
+    /**
+     * Creates a new outbound/local stream.
+     */
+    DefaultHttp2FrameStream newStream(int streamId) throws Http2Exception {
+        assert ctx.executor().inEventLoop();
```

Comment:
from users point of view this requirement might be rather unexpected for *creating* stream as opposed to writing message to pipeline

Reply:
you mean the `assert` ?